app/services/declarations/create.rb

```diff
@@ -0,0 +1,166 @@
+# frozen_string_literal: true
+
+module Declarations
+  class Create
+    include ActiveModel::Model
+    include ActiveModel::Attributes
+
+    attribute :lead_provider
+    attribute :participant_id
+    attribute :declaration_type, :string
+    attribute :declaration_date, :datetime
+    attribute :course_identifier, :string
+    attribute :has_passed
+
+    validates :lead_provider, presence: true
+    validates :participant_id, presence: true
+    validates :participant, participant_presence: true, participant_not_withdrawn: true
+    validates :course_identifier, inclusion: { in: Course::IDENTIFIERS }, allow_blank: false, course_for_participant: true
+    validates :declaration_date, declaration_date: true
+    validates :declaration_date, presence: true
+    validates :declaration_type, presence: true
+    # TODO: we don't have NPQ Contract yet
+    # validates :cohort, contract_for_cohort_and_course: true
+
+    validate :output_fee_statement_available
+    validate :validate_has_passed_field, if: :validate_has_passed?
+    validate :validate_schedule_exists
+    validate :validates_billable_slot_available
+
+    attr_reader :raw_declaration_date, :declaration
+
+    def create_declaration
+      return false unless valid?
+
+      ApplicationRecord.transaction do
+        find_or_create_declaration!
+        set_eligibility!
+
+        statement_attacher.attach unless declaration.submitted_state?
+      end
+
+      true
+    end
+
+    def declaration_date=(raw_declaration_date)
+      self.raw_declaration_date = raw_declaration_date
+      super
+    end
+
+    def application
+      @application ||= participant
+        &.applications
+        &.accepted
+        &.includes(:course)
+        &.find_by(lead_provider:, course: { identifier: course_identifier })
+    end
+
+    def participant
+      @participant ||= begin
+        Participants::Query.new(lead_provider:).participant(ecf_id: participant_id)
+      rescue StandardError
+        nil
+      end
```

Comment:
This is really dangerous.

Reply:
Agree.. I've made the rescue exception specific.. tks 👍